File: src/lib/assets.ts
// ---------------------------------------------------------------------------
// Static asset URLs that must survive a non-root deployment base.
// ---------------------------------------------------------------------------
// The brand logo lives in public/, so Vite copies it verbatim and does NOT
// rewrite references to it — a hardcoded "/flyer-logo.png" stays absolute in the
// built bundle. That is correct for the web deploy (served from a domain root)
// and broken for the desktop build, which loads dist/index.html over file://:
// there, "/flyer-logo.png" resolves against the FILESYSTEM root, so every place
// the logo appears (sidebar, welcome screen, chat header, assistant avatar)
// silently 404s with ERR_FILE_NOT_FOUND and the app looks unbranded.
//
// import.meta.env.BASE_URL is Vite's build-time `base`, so this resolves to
// "/flyer-logo.png" on the web (byte-identical to before) and "./flyer-logo.png"
// under the desktop build's base:"./" — relative to dist/index.html, which is
// where the file actually is.
//
// One constant rather than four inline expressions: the four call sites are in
// four different components, and a per-site fix is exactly the kind of thing
// that gets half-applied when a fifth site appears.
export const LOGO_URL = `${import.meta.env.BASE_URL}flyer-logo.png`;

File: android/app/src/main/java/com/hypersys/lib/Assets.kt
package com.hypersys.lib

import android.content.Context
import androidx.annotation.DrawableRes

/**
 * Port of src/lib/assets.ts.
 *
 * The web file exists because Vite does not rewrite absolute references in
 * public/ assets, so a hardcoded "/flyer-logo.png" breaks under a non-root
 * deployment base (the desktop build loads over file://). One constant instead
 * of four inline expressions, because a per-site fix is exactly the kind of
 * thing that gets half-applied when a fifth site appears.
 *
 * Android's equivalent problem is resource access, and its answer is a layer
 * the web does not have: R.drawable.* — compile-time-verified, base-safe by
 * construction, no URL at all. What survives the port is therefore the *rule*,
 * not the string: every place the logo appears resolves through this one
 * constant, so a future asset swap (a themed variant, a density split) is a
 * one-line change rather than a five-file hunt.
 */
object Assets {

    /** The brand logo, everywhere it appears. */
    @DrawableRes
    const val LOGO_RES: Int = R.drawable.ic_launcher_foreground

    /** Debug label mirroring LOGO_URL's name in the web tree. */
    fun logoName(): String = "flyer-logo"
}

/** Convenience for the call sites that just need the drawable id. */
@DrawableRes
fun logoRes(): Int = Assets.LOGO_RES

/** Debug-only placeholder resolving the resource to its entry name. */
fun Context.logoEntryName(): String =
    runCatching { resources.getResourceEntryName(Assets.LOGO_RES) }.getOrDefault("flyer-logo")
